presto-orc/src/main/java/com/facebook/presto/orc/OrcSelectiveRecordReader.java

```diff
@@ -428,6 +450,14 @@ private void initializePositions(int batchSize)
         }
     }
 
+    private int applyFilterFunctionsWithNoInputs(int[] positions, int positionCount)
+    {
+        initializeOutputPositions(positionCount);
+        Page page = new Page(positionCount);
+        positionCount = filterFunctionWithoutInput.get().filter(page, positions, positionCount, errors);
```

Comment:
- avoid re-assigning input parameters; 

`return filterFunctionWithoutInput.get().filter(page, positions, positionCount, errors);`

Reply:
When you say positions, do you mean we can use positions instead of outputPositions?